keyboards/kemove/snowfox/keymaps/default/keymap.c

```diff
@@ -126,7 +126,7 @@ const uint16_t keymaps_size = sizeof(keymaps);
 void dip_switch_update_user(uint8_t index, bool active) {
   switch (index) {
     case 0:
-      if (active) {
+      if (!active) {
```

Comment:
Wouldn't this also affect other revision's function?

Reply:
I think you are correct. Then the second question would be how to manage this difference within the abstraction layer. I would propose leveraging the inverse function built into GPIOs by complementing the HAL header in `chibios-contrib` to add the following macros:
```
#define   MODE_HYS_POS            5U
#define   MODE_HYS_DISABLE        (0x0U << MODE_HYS_POS)
#define   MODE_HYS_ENABLE         (0x1U << MODE_HYS_POS)

#define   MODE_INV_POS            6U
#define   MODE_INV_DISABLE        (0x0U << MODE_INV_POS)
#define   MODE_INV_ENABLE         (0x1U << MODE_INV_POS)
```

So that the inverse logic is dealt within the board initialization code:
```
    palSetLineMode(LINE_WINMAC, MODE_DIR_IN | MODE_MODE_PULL_UP | MODE_INV_ENABLE | MODE_AD_DIGITAL);
```

Does this sound feasible?

Reply:
I think the solution you proposed is quite elegant

You can make the change to ChibiOS if you'd like and test it locally make sure it works. I've never tested the INVERT function before. But I'd assume it works. 

If it does not work, I can try and read QMK to see if there is something that might help us in this case.

Reply:
I had a local build with the proposal above, and fortunately it works pretty well.

I believe QMK should have such an abstracted interface for such a thing, but if our journey can stop here then I wouldn't other too much about it.

Reply:
I think it is fine to leave it like this if it works.

Reply:
One thing just came across my mind: Can we confirm that the current logic works across v1.2 v1.5 and v3 except v1.4? That sounds a bit weird to me TBH

Reply:
Yeah this is interesting, can't say anything for 1.5, but I have 1.2 and v3, they both work correctly

Reply:
Well if that's the case then I'll push the change. Am I supposed to create another PR on `chibios-contrib` repo?

Reply:
I think that needs to happen first.

Reply:
It's opened [here](https://github.com/OpenKemove/ChibiOS-Contrib/pull/1), but for some reason I couldn't add you as a reviewer.